app/models/user.rb

```diff
@@ -1,2 +1,17 @@
 class User < ActiveRecord::Base
+  has_secure_password
+  has_many :questions, foreign_key: 'creator_id'
+  has_many :answers
+  validates :email, presence: true
+  validates :email, format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i, on: :create }
```

Comment:
You probably don't need the on: :create here. It will only run when the record is first created - you may well want to prevent bad updates as well.